lib/src/runner/browser/browser.dart

```diff
@@ -126,6 +126,8 @@ abstract class Browser {
     });
 
     // Swallow exceptions. The user should explicitly use [onExit] for these.
-    return onExit.catchError((_) {});
+    // However, surface application exceptions which are thrown due to startup
+    // issues with the browser.
+    return onExit.catchError((_) {}, test: (e) => e is! ApplicationException);
```

Comment:
> The close errors happen in a different zone than what is seen in the BrowserManager.

I don't think that's true. There was a zone difference in the test you were running that interacted with `Browser` on its own, but that was specific to the test environment you were using—I think it's unrelated to the bug that's causing flakiness on Travis. If you want to test that, you'll probably need to invoke `test` as an executable that loads a browser.